packages/runner/src/storage/interface.ts

```diff
@@ -255,51 +269,116 @@ export interface IStorageTransactionAborted extends Error {
  * Error indicates that transaction consistency guarantees have being
  * invalidated - some fact has changed while transaction was in progress.
  */
-export interface IStorageConsistencyError extends Error {}
+export interface IStorageTransactionInconsistent extends Error {
+  name: "StorageTransactionInconsistent";
+}
+
+/**
+ * Error that indicating that no change could be made to a transaction is it is
+ * no longer active.
+ */
+export type InactiveTransactionError =
+  | IStorageTransactionInconsistent
+  | IStorageTransactionAborted
+  | IStorageTransactionFailed
+  | IStorageTransactionComplete;
 
 export type IStorageTransactionError =
   | IStorageTransactionAborted
+  | IStorageTransactionInconsistent
+  | IStorageTransactionFailed;
+
+export type IStorageTransactionFailed =
   | ConflictError
   | TransactionError
   | ConnectionError
   | AuthorizationError;
 
-export interface IStorageTransactionClosed extends Error {}
+export type IReaderError =
+  | IStorageTransactionComplete
+  | IStorageTransactionAborted;
+
+export type IWriterError =
+  | IStorageTransactionComplete
+  | IStorageTransactionAborted
+  | IStorageTransactionInconsistent
+  | IStorageTransactionWriteIsolationError;
+
+export interface IStorageTransactionComplete extends Error {
+  name: "StorageTransactionCompleteError";
+}
 export interface INotFoundError extends Error {}
 export type IStorageTransactionProgress = Variant<{
   open: IStorageTransactionLog;
```

Comment:
clarification: to entries move from `open` to `pending` to `done`?

if i want to know _all_, no matter the status (as scheduler needs it), should it iterate over all of them? who else is going to call this?